packages/common/e2e/dynamic-value-resolver.e2e.ts

```diff
@@ -0,0 +1,226 @@
+}
+
+interface ICreateOutput {
+  initiative: HubInitiative;
+  projects: HubProject[];
+}
+
+async function createItems(
+  childCount: number,
+  includeFSQuery: boolean,
+  context: IArcGISContext
+): Promise<ICreateOutput> {
+  const result: ICreateOutput = {
+    // THIS IS TERRIBLE! DO NOT DO THIS IN PRODUCTION CODE!
+    initiative: null as unknown as HubInitiative,
+    projects: [],
+  };
+  // Create an initiative
+  const init: Partial<IHubInitiative> = {
+    name: "Test Initiative",
+    description: "This is a test initiative",
+    orgUrlKey: context.portal.urlKey,
+  };
+  const instance = await HubInitiative.create(init, context, true);
+  // Construct typed dynamic value definitions
+  const treesDynamicValue: IDynamicItemQueryDefinition = {
+    source: "item-query",
+    sourcePath: "properties.values.treesPlanted",
+    outPath: "treesPlanted",
+    options: {
+      q: `type: "Hub Project" AND typekeywords: initiative|${instance.id}`,
+    },
+    aggregation: "sum",
+    required: false,
+  };
+  const statusDynamicValue: IDynamicItemQueryDefinition = {
+    source: "item-query",
+    sourcePath: "properties.values.status",
+    outPath: "statusCounts",
+    options: {
+      q: `type: "Hub Project" AND typekeywords: initiative|${instance.id}`,
+    },
+    aggregation: "countByValue",
+    required: true,
+  };
+  const cascadeDynamicValue: IDynamicItemQueryDefinition = {
+    source: "item-query",
+    options: {
+      q: `type: "Hub Project" AND typekeywords: initiative|${instance.id}`,
+    },
+    sourcePath: "properties.values.surveyCount",
+    outPath: "surveyCounts",
+    aggregation: "sum",
+    required: false,
+  };
+
+  const dynamicValues = [
+    treesDynamicValue,
+    statusDynamicValue,
+    cascadeDynamicValue,
+
+    {
+      source: "portal",
+      sourcePath: "urlKey",
+      outPath: "orgShort",
+    } as IDynamicPortalQueryDefinition,
+    {
+      source: "portal",
+      sourcePath: "name",
+      outPath: "orgName",
+    } as IDynamicPortalQueryDefinition,
+    {
+      source: "portal",
+      sourcePath: "defaultExtent",
+      outPath: "extent",
+    } as IDynamicPortalQueryDefinition,
+  ];
+  // This is icky... makes me consider adding methods to the class
+  const json = instance.toJson();
+  json.dynamicValues = dynamicValues;
+  instance.update(json);
+  await instance.save();
+  result.initiative = instance;
+  // add 2 metrics to the initiative
+  // Create N projects connected to the initiative
+  const connectingKeyword = `initiative|${instance.id}`;
+  // assign values to tne metrics for each project
+  for (let i = 0; i < childCount; i++) {
+    const prj: Partial<IHubProject> = {
+      name: `Test Project ${i}`,
+      description: "This is a test project",
+      orgUrlKey: context.portal.urlKey,
+      typeKeywords: [connectingKeyword, "HubProject"],
+      values: {
+        treesPlanted: i * 10,
+        status: i % 2 === 0 ? "planned" : "started",
```

Comment:
should this enumeration be a function? or how would this be handled when there are multiple status enuemrations?

Reply:
this line is just switching up values in the fake data that's generated for the tests. For actual categorical metric fields, we need to decide if we want the Parent to be able to declare the valid values for the DataValue from which the Child entity must choose (we can do this via the UI Schema - more a question of how we store that in the definition structure)